src/neo/Ledger/MemoryPool.cs

```diff
@@ -306,12 +308,14 @@ internal VerifyResult TryAdd(Transaction tx, DataCache snapshot)
                 _txRwLock.ExitWriteLock();
             }
 
-            foreach (IMemoryPoolTxObserverPlugin plugin in Plugin.TxObserverPlugins)
-            {
-                plugin.TransactionAdded(_system, poolItem.Tx);
-                if (removedTransactions != null)
-                    plugin.TransactionsRemoved(_system, MemoryPoolTxRemovalReason.CapacityExceeded, removedTransactions);
-            }
+            TransactionAdded?.Invoke(this, poolItem.Tx);
+            if (removedTransactions != null)
+                foreach (Transaction removed in removedTransactions)
+                    TransactionRemoved?.Invoke(this, new()
+                    {
+                        Transaction = removed,
+                        Reason = TransactionRemovalReason.CapacityExceeded
+                    });
```

Comment:
Use the array in the event?, if the plugin want to iterate, it should be his responsibility

```suggestion
                  TransactionRemoved?.Invoke(this, new()
                  {
                      Transactions = removedTransactions,
                      Reason = TransactionRemovalReason.CapacityExceeded
                  });
```

Reply:
But `TransactionAdded` doesn't use array.

Reply:
I know, but it's more optimal to use an array for remove.

Reply:
Done.